cli/src/auth/okta.rs

```diff
@@ -309,3 +318,43 @@ pub async fn refresh_tokens(config: &Config) -> Result<OktaAuth, WKCliError> {
         refresh_token: new_refresh_token.secret().to_owned(),
     })
 }
+
+pub async fn introspect_token(config: &Config, token: &str) -> Result<bool, WKCliError> {
+    let okta_client_id = ClientId::new(config.core.okta_client_id.clone());
+
+    let issuer_url =
+        IssuerUrl::new("https://mindvalley.okta.com".to_string()).expect("Invalid issuer URL");
+
+    // Fetch Okta's OpenID Connect discovery document.
+    let provider_metadata = CoreProviderMetadata::discover_async(issuer_url, async_http_client)
+        .await
+        .map_err(|_err| AuthError::OpenIDDiscoveryError)?;
+
+    // Set up the config for the Okta OAuth2 process.
+    let client = CoreClient::from_provider_metadata(provider_metadata, okta_client_id, None)
+        .set_introspection_uri(
+            IntrospectionUrl::new("https://mindvalley.okta.com/oauth2/v1/introspect".to_string())
+                .expect("Invalid introspect URL"),
+        );
+
+    let token_response = client
+        .introspect(&AccessToken::new(token.to_owned()))
+        .map_err(|_err| AuthError::OpenIDConnectError {
+            message: "Failed to contact introspect endpoint - 1".to_string(),
+        })?
+        .set_token_type_hint("refresh_token")
+        .request_async(async_http_client)
+        .await
+        .map_err(|_err| AuthError::OpenIDConnectError {
+            message: "Failed to get introspect response".to_string(),
+        })?;
+
+    debug!(
```

Comment:
If the refresh token is expired, we can't get the following fields
* username
* exp
Only the active field is available, so we will need to handle it otherwise it will be a panic in the client side. I think you should add a test to check this behaviour.

Reply:
This is example from Okta's documentation 

![image](https://github.com/mindvalley/wukong-cli/assets/1546109/73555768-3a4b-441a-9c09-ff008e814ab4)

Reply:
This handling isn't require because these values are optional fields in the response other then active.

Tested with invalid and here is the result:
`introspect response: refresh_token: asdfsadfasa, email: None, active: false, exp: None`